model/sessions.js

```diff
@@ -0,0 +1,11 @@
+var mongoose = require("mongoose");
+var Schema = mongoose.Schema;
+let sessionSchema = new Schema({
+  sessionId: String,
+  userId: Number,
+  provider: String,
+  createdAt: Number,
+  active: Boolean,
+});
+
+module.exports = mongoose.model("sessions", sessionSchema);
```

Comment:
Why we are using session on server side . I think we shouldn't complicate our arch just let's keep simple since we are jwt based authentication why to maintain session / cookies. As per my understanding all this should be maintained on client side not on. server side.

Reply:
according to the best architecture in case of transactional application control of session has to be done from backed not from client side.

Reply:
I think there is a way we can avoid using sessions for authentication , I am fine with sessions for rate limiting. We can discuss more on this

Reply:
session is not for ratelimting prefereabbly it for forefull session expiry control from backend